cura/API/Machines.py

```diff
@@ -0,0 +1,139 @@
+            if "group_id" in metadata:
+                fake_machine.group_id = global_stack.getMetaDataEntry("group_id")
+            if "group_name" in metadata:
+                fake_machine.group_name = global_stack.getMetaDataEntry("group_name")
+            if "um_network_key" in metadata:
+                fake_machine.um_network_key = global_stack.getMetaDataEntry("um_network_key")
+
+            fake_machine.connection_types = global_stack.configuredConnectionTypes
+            
+        return fake_machine
+
+    ##  Set the current machine's friendy name.
+    #   This is the same as "group name" since we use "group" and "current machine" interchangeably.
+    #   TODO: Maybe make this "friendly name" to distinguish from "hostname"?
+    @pyqtSlot(str)
+    def setCurrentMachineGroupName(self, group_name: str) -> None:
+        Logger.log("d", "Attempting to set the group name of the active machine to %s", group_name)
+        global_stack = self._application.getGlobalContainerStack()
+        if global_stack:
+            # Update a GlobalStacks in the same group with the new group name.
+            group_id = global_stack.getMetaDataEntry("group_id")
+            machine_manager = self._application.getMachineManager()
+            for machine in machine_manager.getMachinesInGroup(group_id):
+                machine.setMetaDataEntry("group_name", group_name)
+
+            # Set the default value for "hidden", which is used when you have a group with multiple types of printers
+            global_stack.setMetaDataEntry("hidden", False)
+
+    ##  Set the current machine's configuration from an (optional) output device.
+    #   If no output device is given, the first one available on the machine will be used.
+    #   NOTE: Group and machine are used interchangeably.
+    #   NOTE: This doesn't seem to be used anywhere. Maybe delete?
+    @pyqtSlot(QObject)
+    def updateCurrentMachineConfiguration(self, output_device: Optional["PrinterOutputDevice"]) -> None:
+
+        if output_device is None:
+            machine_manager = self._application.getMachineManager()
+            output_device = machine_manager.printerOutputDevices[0]
+        
+        hotend_ids = output_device.hotendIds
+        for index in range(len(hotend_ids)):
+            output_device.hotendIdChanged.emit(index, hotend_ids[index])
+        
+        material_ids = output_device.materialIds
+        for index in range(len(material_ids)):
+            output_device.materialIdChanged.emit(index, material_ids[index])
+
+    ##  Add an output device to the current machine.
+    #   In practice, this means:
+    #   - Setting the output device's network key in the current machine's metadata
+    #   - Adding the output device's connection type to the current machine's configured connection
+    #     types.
+    #   TODO: CHANGE TO HOSTNAME
+    @pyqtSlot(QObject)
+    def addOutputDeviceToCurrentMachine(self, output_device: "PrinterOutputDevice") -> None:
```

Comment:
It wont, since no one else uses this function. we know its only serving the purpose of one plugin right now because we only moved code around as part of the process of figuring out what exactly we need and want to build, with the assumption of more prs incoming soon as we make this usable in more than a single circumstance.

Reply:
The intent is to have a single entry point, right? So other features will need to change to use this function as well. Nobody is using it since it got merged a few days ago. I understand that it's not your responsibility to make sure  that Cura works for other connection methods and future circumstances, but it is mine.

It was intended that this functionality would be implemented by the individual plug-ins anyway. That was the architecture design we've discussed a few months ago with you. The plug-in gets a callback to store the output device in a newly added printer so that it can implement its plug-in specific behaviour.